security/s2a/internal/crypter/counter_test.go

```diff
@@ -34,75 +39,66 @@ func TestCounterInc(t *testing.T) {
 			overflow: true,
 		},
 	} {
-		t.Run(test.desc, func(t *testing.T) {
-			// Make first getAndIncrement call. This should return the same value
-			// which was given.
-			c := newCounter(test.counter)
-			value, err := c.getAndIncrement()
-			if err != nil {
-				t.Errorf("counter(%v).getAndIncrement() returned error: %v", test.counter, err)
-			}
-			if value != test.counter {
-				t.Errorf("counter(%v).getAndIncrement() = %v, want %v", test.counter, value, test.counter)
-			}
-			if test.overflow {
-				if !c.hasOverflowed {
-					t.Errorf("counter(%v).getAndIncrement() did not set hasOverflowed flag", test.counter)
-				}
-			} else {
-				if c.hasOverflowed {
-					t.Errorf("counter(%v).getAndIncrement() unexpectedly set hasOverflowed flag", test.counter)
-				}
-			}
-
-			// Make second getAndIncrement call. This should verify that the first
-			// getAndIncrement call succeeded in incrementing the value.
-			value, err = c.getAndIncrement()
-			if test.overflow {
+		t.Run(tc.desc, func(t *testing.T) {
+			c := counterFromValue(tc.counter)
```

Comment:
There's no need for a separate function for this. You can do:

```
c := counter{value: tc.counter}
```

Reply:
Done